rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,537 @@
+        for entry in balances:
+            try:
+                symbol = entry['token']
+            except KeyError as e:
+                log.error(
+                    'Error processing a Woo balance.',
+                    entry=entry,
+                    error=str(e),
+                )
+                self.msg_aggregator.add_error(
+                    'Symbol could not be found in Woo balance entry.'
+                    'Check logs for details. Ignoring it.',
+                )
+                continue
+            try:
+                if (amount := deserialize_asset_amount(entry['holding'] + entry['staked'])) == ZERO:  # noqa: E501
+                    continue
+                asset = asset_from_woo(symbol)
+            except (DeserializationError, KeyError) as e:
+                log.error(
+                    'Error processing a Woo balance.',
+                    entry=entry,
+                    error=str(e),
+                )
+                self.msg_aggregator.add_error(
+                    f'Failed to deserialize a Woo balance entry for symbol {symbol}.'
+                    'Check logs for details. Ignoring it.',
+                )
+                continue
+            except UnknownAsset as e:
+                log.error(str(e))
+                self.msg_aggregator.add_warning(
+                    f'Found unknown Woo asset {e.identifier}. Ignoring it in the balance query.',
+                )
+                continue
+            try:
+                usd_price = Inquirer().find_usd_price(asset=asset)
+            except RemoteError as e:
+                log.error(str(e))
+                self.msg_aggregator.add_error(
+                    f'Error processing Woo balance result due to inability to '
+                    f'query USD price: {e}. Skipping balance entry.',
+                )
+                continue
+
+            assets_balance[asset] += Balance(
+                amount=amount,
+                usd_value=amount * usd_price,
+            )
+        return dict(assets_balance)
+
+    def _deserialize_trade(self, trade: dict[str, Any]) -> Trade:
+        """
+        Deserialize a Woo trade returned from the API to History Event
+
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        symbol = trade['symbol']
+        try:
+            _, base_asset_symbol, quote_asset_symbol = symbol.split('_')
+        except ValueError as e:
+            raise DeserializationError(
+                f'Could not split symbol {symbol} into base and quote asset',
+            ) from e
+        base_asset = asset_from_woo(base_asset_symbol)
+        quote_asset = asset_from_woo(quote_asset_symbol)
+        side = trade['side']
+        fee_asset = asset_from_woo(trade['fee_asset'])
+        trade_type = TradeType.BUY if side == 'BUY' else TradeType.SELL
+        timestamp = deserialize_timestamp(int(float(trade['executed_timestamp'])))
```

Comment:
why the `int(float(x))` here? Is it returned as a string? If yes just pass it directly to the function